### DISCUSSION TITLE
How to use constants in forms?

### DISCUSSION THREAD

**Discussion by checkifexist** [ansibleguy76/ansibleforms#144] [Q&A] [answered]
Hi devs,

In Designer there is section for "Constants", how to use them? an example would help
Are these Constant Individual form level or global?
If it is global, can we set this from one form field and it can be use in other form?

**Comment by ansibleguy76** [comment]
A constant is available in every form.  it's fixed and cannot be overwritten.

(https://ansibleforms.com/forms/#formsyaml_constants)

**Reply by checkifexist** [reply]
I Added example constants here
![image](https://github.com/ansibleguy76/ansibleforms/assets/157472645/0cd02a06-16b0-4be4-b8a0-7717523b9e29)
and is using here
![image](https://github.com/ansibleguy76/ansibleforms/assets/157472645/52f1a930-c20c-4aa1-b645-2b73b687734f)
but its not printing value, rather its printing this
![image](https://github.com/ansibleguy76/ansibleforms/assets/157472645/5a02036b-8df1-469a-a5bd-d33319e1d030)

**Reply by ansibleguy76** [reply]
nono :

constants are variables you can use in expressions.  Field names are never dynamic.

like : 
expression: "fn.fnRestBasic('get','https://$(SERVERX)/....

**Reply by checkifexist** [reply]
okay got it,

tried it in ssh but its showing following warning
![image](https://github.com/ansibleguy76/ansibleforms/assets/157472645/86cc376f-cafb-4930-bf39-25126f3178f3)

i used this expression 
![image](https://github.com/ansibleguy76/ansibleforms/assets/157472645/e4223fa5-0572-4d69-b57c-4a598a98c8e4)
constant 
![image](https://github.com/ansibleguy76/ansibleforms/assets/157472645/2c381ff7-20b5-4442-9908-da90c2928680)

**Reply by ansibleguy76** [reply]
Don't forget quote's.  You simple assemble a command.  
expression: fn.fnSsh('user','$(SERVER_IP)','ls')

**Reply by checkifexist** [reply, marked as answer]
nvm,  '$(SERVER_IP)' fixed it, thx
